### PR TITLE
Global key support

### PR DESCRIPTION
**Pull request by jasoncheng-jora**
- Vocab push: push custom key to Phrase
- Vocab pull: use translation of custom key
- Add Vocab CLI feature flag --error-on-no-global-key-translation
- Doc: Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -506,6 +506,37 @@ Tags on keys in other languages will be ignored.
...
+```
+
+In the above example,
+
+- `vocab push` will push the `hello` and `app.goodbye.label` keys to Phrase.
+- `vocab pull` will pull translations from Phrase and map them to the `hello` and `app.goodbye.label` keys.
+
+##### Error on no translation for global key
+
+We can throw error if Phrase does not have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`. E.g.
```
Comment (rstacruz) [inline README.md:534]:
```suggestion
An error can be thrown if Phrase doesn't have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`.
```

**suggestion:** might be worth refactoring the sentence to remove the ambiguous "we" (who's we? vocab? the vocab team?).

**Review by rstacruz — COMMENTED** [review]
I'll leave it to the @askoufis @jahredhope for the full review, but this looks good so far :+1: Thanks for the comprehensive update to the tests and docs as well.

File: README.md (hunk 1 above)
Comment (jasoncheng-jora) [reply README.md]: @rstacruz That's a good call-out, appreciate it

**Review by adriankencana — COMMENTED** [review]
Looking good 👍🏻

File: .changeset/nasty-seahorses-confess.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Vocab pull: support custom key
+
+```json
+{
+  "thanks": {
+    "message": "Thanks",
+    "global-key": "app.thanks.label"
+  }
+}
+```
```
Comment (askoufis) [inline .changeset/nasty-seahorses-confess.md:15]: Generally we try and put a bit more detail/description into the changeset, as well as an example if appropriate so that consumers don't _have_ to read the README to understand what the new API does and how to use it.

File: README.md [hunk 3]
```diff
@@ -506,6 +506,37 @@ Tags on keys in other languages will be ignored.
...
+```
+
+In the above example,
+
+- `vocab push` will push the `hello` and `app.goodbye.label` keys to Phrase.
+- `vocab pull` will pull translations from Phrase and map them to the `hello` and `app.goodbye.label` keys.
+
+##### Error on no translation for global key
+
+Vocab can be configured to throw an error if Phrase doesn't have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`.
```
Comment (askoufis) [inline README.md:534]:
```suggestion
By default, `vocab pull` will not error if a translation is missing in Phrase for a translation with a global key.
If you want to throw an error in this situation, pass the `--error-on-no-global-key-translation` flag:
```

**Review by askoufis — COMMENTED** [review]
Thanks for the PR. I've got some feedback around naming and release notes/docs, but overall the PR looks good.

File: packages/phrase/src/config.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const GLOBAL_KEY = 'global-key';
```
[the remark this answers is not in the document]
Comment (jasoncheng-jora) [reply packages/phrase/src/config.ts:1]: Removing this constant. Since we are renaming global key to camel case, it is much easier to access it without having to use index.

File: packages/core/src/config.ts [hunk 5]
```diff
@@ -202,3 +202,5 @@ export function resolveConfigSync(
 
   return null;
 }
+
+export const GLOBAL_KEY = 'global-key';
```
[the remark this answers is not in the document]
Comment (jasoncheng-jora) [reply packages/core/src/config.ts:206]: Removed.

File: packages/core/src/load-translations.ts [hunk 6]
```diff
@@ -410,6 +411,17 @@ export async function loadAllTranslations(
         );
       }
       keys.add(uniqueKey);
+
+      const globalKey =
+        loadedTranslation.languages[config.devLanguage][key][GLOBAL_KEY];
+      if (globalKey) {
+        if (keys.has(globalKey)) {
+          throw new Error(
+            `Duplicate keys found. Global key ${key} was found multiple times`,
```
[the remark this answers is not in the document]
Comment (jasoncheng-jora) [reply packages/core/src/load-translations.ts]: Agree

**Comment by jasoncheng-jora** [comment]
Thanks for the review @askoufis

File: .changeset/many-apricots-try.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---
+'@vocab/phrase': minor
+'@vocab/cli': minor
+---
```
Comment (askoufis) [inline .changeset/many-apricots-try.md:4]: The `@vocab/phrase` change should be moved into a separate changeset as it requires different release notes. `phrase` is an API, so release notes for the CLI aren't relevant.

Take a look at [a previous `@vocab/phrase` release](https://github.com/seek-oss/vocab/releases/tag/%40vocab%2Fphrase%401.1.0) to get an idea of what to write in the changeset.

Comment (jasoncheng-jora) [reply .changeset/many-apricots-try.md]: Thanks for the heads-up, makes sense!

**Comment by askoufis** [comment]
FYI I published some snapshots of the changes on this branch [here](https://github.com/seek-oss/vocab/actions/runs/6646935333) if you want to try them in an app. Just updating regular dependencies might not update all versions of the package, so you'll probably want to use `resolutions`/`overrides` as some packages depend on eachother.

**Comment by jasoncheng-jora** [comment]
@askoufis Awesome, thanks.

**Comment by jasoncheng-jora** [comment]
@askoufis Tested in Echo web app and it works correctly.

**Review by askoufis — APPROVED** [review]
